honeybee_energy/simulation/runperiod.py

```diff
@@ -0,0 +1,363 @@
+        return self._start_date
+
+    @start_date.setter
+    def start_date(self, value):
+        if value is not None:
+            self._check_date(value, 'start_date')
+            self._start_date = value
+        else:
+            self._start_date = Date(1, 1)
+        self._check_start_before_end()
+
+    @property
+    def end_date(self):
+        """Get or set a ladybug Date object for the end of the run period."""
+        return self._end_date
+
+    @end_date.setter
+    def end_date(self, value):
+        if value is not None:
+            self._check_date(value, 'start_date')
+            self._end_date = value
+        else:
+            self._end_date = Date(12, 31)
+        self._check_start_before_end()
+
+    @property
+    def start_day_of_week(self):
+        """Get or set text for the day of the week on which the simulation starts.
+
+        Choose from the following:
+            * Sunday
+            * Monday
+            * Tuesday
+            * Wednesday
+            * Thursday
+            * Friday
+            * Saturday
+        """
+        return self._start_day_of_week
+
+    @start_day_of_week.setter
+    def start_day_of_week(self, value):
+        clean_input = valid_string(value).lower()
+        for key in self.DAYS_OF_THE_WEEK:
+            if key.lower() == clean_input:
+                value = key
+                break
+        else:
+            raise ValueError(
+                'start_day_of_week {} is not recognized.\nChoose from the '
+                'following:\n{}'.format(value, self.DAYS_OF_THE_WEEK))
+        self._start_day_of_week = value
+
+    @property
+    def holidays(self):
+        """Get or set a list of ladybug Date objects for holidays."""
+        return self._holidays
+
+    @holidays.setter
+    def holidays(self, value):
+        if value is not None:
+            if not isinstance(value, tuple):
+                value = tuple(value)
+            for date in value:
+                assert isinstance(date, Date), 'Expected ladybug Date for ' \
+                    'RunPeriod holiday. Got {}.'.format(type(date))
+        self._holidays = value
+
+    @property
+    def daylight_saving_time(self):
+        """Get or set a DaylightSavingTime object for start and end of daylight savings.
+        """
+        return self._daylight_saving_time
+
+    @daylight_saving_time.setter
+    def daylight_saving_time(self, value):
+        if value is not None:
+            assert isinstance(value, DaylightSavingTime), 'Expected DaylightSavingTime' \
+                ' for RunPeriod run_period. Got {}.'.format(type(value))
+        self._daylight_saving_time = value
+
+    @property
+    def is_leap_year(self):
+        """Get or set a boolean noting whether the RunPeriod is for a leap year simulation.
+        """
+        return self.start_date.leap_year
+
+    @is_leap_year.setter
+    def is_leap_year(self, value):
```

Comment:
You are right that EneryPlus accepts year inputs now in the RunPeriod definition but there are a number of really good reasons why we should not support it:
1. OpenStudio does not support it and, as far as I understand, there are no plans to support it in the future.
2. It would create a coordination nightmare between ladybug and honeybee-energy. To just scratch the surface of the headache that this will create, someone running a multi year energy simulation in E+ will not be able to import their results to ladybug data collections, making it impossible to work with the results in ladybug.
3. If you want to go through the whole effort of exposing year properties across ladybug datetimes and analysis periods to make this work, then you are welcome but this really is an example of letting an edge case drive the core development.  Realistically, the only reason why you'd want to make use of the year input on the RunPeriod is to run a multi-year simulation and, if someone badly needs this, then I don't think it's that big of deal to ask users to just break up their climate data into one year per EPW and run separate simulations.

Reply:
The reason 1 is good enough. I think we wait and see when OpenStudio is going to support this.